### PR TITLE
RPC Pending Transactions Filter

### PR DESCRIPTION
**Pull request by tomusdrw**
Is separating `transaction_data` in `PollManager` a good idea?  Maybe it should be just merged to `PollInfo`?
Actually for pending transactions filter we don't really need `block_number`, but for blocks filter we don't need `transactions_data`

@debris Could you have a look?

### CODE REVIEW

File: rpc/src/v1/helpers/poll_manager.rs [hunk 1]
```diff
@@ -57,15 +60,25 @@ impl<F, T> PollManager<F, T> where T: Timer {
...
 		PollManager {
 			polls: TransientHashMap::new_with_timer(POLL_LIFETIME, timer),
+			transactions_data: HashMap::new(),
 			next_available_id: 0,
 		}
 	}
 
+	fn prune(&mut self) {
+		self.polls.prune();
+		// self.polls.prune()
```
Comment (tomusdrw) [inline rpc/src/v1/helpers/poll_manager.rs]: Waiting for new version of `TransientHashMap`

**Comment by debris** [comment]
> Is separating transaction_data in PollManager a good idea? Maybe it should be just merged to PollInfo?

I think, neither of these two solutions. You can store hashes in [`PollFilter::PendingTransaction`](https://github.com/ethcore/parity/blob/master/rpc/src/v1/helpers/poll_filter.rs#L8) field :) And `PollFilter` is already a part of `PollInfo` so it won't require any additional changes besides this one.

**Comment by debris** [comment]
> Actually for pending transactions filter we don't really need block_number, but for blocks filter we don't need transactions_data

You are right... `block_number` could be just a field of `PollFilter::Block` and `PollFilter::Logs`. And we could completely get rid of `PollInfo` then. This change would also require several small changes in `PollManager` api, but I think it's sensible. What do you think about it?

**Comment by tomusdrw** [comment]
Updated. Looks pretty nice now :)

**Comment by tomusdrw** [comment]
Updated.

**Pull request merged**